src/topology/algebra/infinite_sum.lean

```diff
@@ -432,6 +456,107 @@ end
 
 end tsum
 
+/-!
+### Sums on subtypes
+
+If `s` is a finset of `α`, we show that the summability of `f` in the whole space and on the subtype
+`univ - s` are equivalent, and relate their sums. For a function defined on `ℕ`, we deduce the
+formula `(∑ i in range k, f i) + (∑' i, f (i + k)) = (∑' i, f i)`, in `sum_add_tsum_nat_add`.
+-/
+section subtype
+variables {s : finset β}
+
+lemma has_sum_subtype_iff :
+  has_sum (λ b : {b // b ∉ s}, f b) a ↔ has_sum f (a + ∑ b in s, f b) :=
+begin
+  let gs := λ b, if b ∈ s then f b else 0,
+  let g := λ b, if b ∉ s then f b else 0,
+  have f_sum_iff : has_sum f (a + ∑ b in s, f b) = has_sum (λ b, g b + gs b) (a + ∑ b in s, f b),
+  { congr,
+    ext i,
+    simp [gs, g],
+    split_ifs;
+    simp },
+  have g_zero : ∀ b ∈ s, g b = 0,
+  { assume b hb,
+    dsimp [g],
+    split_ifs,
+    refl },
+  have gs_sum : has_sum gs (∑ b in s, f b),
+  { have : (∑ b in s, f b) = (∑ b in s, gs b),
+    { apply sum_congr rfl (λ b hb, _),
+      dsimp [gs],
+      split_ifs,
+      { refl },
+      { exact false.elim (h hb) } },
+    rw this,
+    apply has_sum_sum_of_ne_finset_zero  (λ b hb, _),
+    dsimp [gs],
+    split_ifs,
+    { exact false.elim (hb h) },
+    { refl } },
+  have : (λ b : {b // b ∉ s}, f b) = (λ b : {b // b ∉ s}, g b),
+  { ext i,
+    simp [g],
+    split_ifs,
+    { exact false.elim (i.2 h) },
+    { refl } },
+  rw [this, has_sum_subtype_iff_of_eq_zero g_zero, f_sum_iff],
+  exact ⟨λ H, H.add gs_sum, λ H, by simpa using H.sub gs_sum⟩,
+end
+
+lemma has_sum_subtype_iff' :
+  has_sum (λ b : {b // b ∉ s}, f b) (a - ∑ b in s, f b) ↔ has_sum f a :=
+by simp [has_sum_subtype_iff]
+
+lemma summable_subtype_iff (s : finset β):
+  summable (λ b : {b // b ∉ s}, f b) ↔ summable f :=
+⟨λ H, (has_sum_subtype_iff.1 H.has_sum).summable, λ H, (has_sum_subtype_iff'.2 H.has_sum).summable⟩
+
+lemma sum_add_tsum_subtype [t2_space α] (s : finset β) (h : summable f) :
+  (∑ b in s, f b) + (∑' (b : {b // b ∉ s}), f b) = (∑' b, f b) :=
+by simpa [add_comm] using
+  has_sum_unique (has_sum_subtype_iff.1 ((summable_subtype_iff s).2 h).has_sum) h.has_sum
+
+/-- Equivalence between the set of natural numbers which are `≥ k` and `ℕ`, given by `n → n - k`. -/
+def not_mem_range_equiv (k : ℕ) : {n // n ∉ range k} ≃ ℕ :=
```

Comment:
I think it would be good to move this to another file. Something like `data.nat.basic` or `data.equiv.algebra`.
And then maybe add the usual simp-lemmas:
* `not_mem_range_equiv_apply`
* `not_mem_range_equiv_symm_apply`.

Reply:
Done